pkg/api/create_person_handler.go

```diff
@@ -0,0 +1,49 @@
+package api
+
+import (
+	"encoding/json"
+
+	"github.com/darchlabs/api-example/pkg/person"
+	"github.com/go-playground/validator/v10"
+)
+
+type CreatePersonHandlerInputs struct {
+	Name string `validate:"required"`
+	Age  int64  `validate:"required"`
+}
+
+type CreatePersonHandler struct {
+	storage PersonStorage
+}
+
+func NewCreatePersonHandler(ps PersonStorage) *CreatePersonHandler {
+	return &CreatePersonHandler{
+		storage: ps,
+	}
+}
+
+func (cp CreatePersonHandler) Invoke(ctx *handlerCtx) *handlerRes {
+	// Validate json schema data
+	validate := validator.New()
+	validate.Struct(ctx.ps)
+
+	// prepare struct for body
+	body := &struct {
+		Person *person.Person `json:"person"`
```

Comment:
Following your implementation y think that you use the `CreatePersonHandlerInputs` type definition

Reply:
But they are different, `CreatePersonHandlerInputs` contains the methods of `PersonStorage` struct, and `Person` is an struct with the fields required by the person object.
Maybe you say is better if I would pass `Person.person` instead of creating an initializing an struct with it?